spd/eval.py

```diff
@@ -755,6 +756,165 @@ def compute(self) -> Mapping[str, float]:
         return {"loss/faithfulness": loss.item()}
 
 
+class GeometricSimilarityComparison(StreamingEval):
+    SLOW = True
+
+    def __init__(self, model: ComponentModel, config: Config, **kwargs: Any):
+        self.model = model
+        self.config = config
+        self.reference_run_path = kwargs.get("reference_run_path")
+        if self.reference_run_path is None:
+            raise ValueError("reference_run_path is required for GeometricSimilarityComparison")
+        self.kwargs = kwargs
+        self.reference_model: ComponentModel | None = None
+        self._computed_this_eval = False
+        self.device = next(iter(model.parameters())).device
+        self.n_tokens = 0
+        self.component_activation_counts: dict[str, Float[Tensor, " C"]] = {
+            module_name: torch.zeros(model.C, device=self.device)
+            for module_name in model.components
+        }
+
+    def _load_reference_model(self) -> ComponentModel:
+        """Load the reference model from wandb or local path"""
+        if self.reference_model is None:
+            from spd.models.component_model import ComponentModel
```

Comment:
This is already imported at the top of the file. I'm surprised the linter doesn't complain here.